src/adaptor/native.ts

```diff
@@ -0,0 +1,30 @@
+import nativeFirestore, {
+  FirebaseFirestoreTypes
+} from '@react-native-firebase/firestore'
+// @ts-ignore: React Native Firebase doesn't export types for internal modules
+import FirestoreDocumentReference from '@react-native-firebase/firestore/lib/FirestoreDocumentReference'
+// @ts-ignore
+import FirestoreFieldValue from '@react-native-firebase/firestore/lib/FirestoreFieldValue'
+// @ts-ignore
+import FirestoreTimestamp from '@react-native-firebase/firestore/lib/FirestoreTimestamp'
```

Comment:
I think instead of this you could use

```
import { firebase } from '@react-native-firebase/firestore';
const FirestoreTimestamp  = firebase.firestore.Timestamp;
```

Not sure if it's compatible with your lib internals, but you don't have to dig into the internals probably

Reply:
That's what I was looking for, thanks!

Reply:
So to check if something is convertable to date you can just

`if (rawInput instanceof firebase.firestore.Timestamp)`